src/test_workflow/smoke_test/smoke_test_cluster_opensearch.py

```diff
@@ -0,0 +1,102 @@
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import shutil
+import time
+
+import requests
+
+from git.git_repository import GitRepository
+from manifests.build_manifest import BuildManifest
+from manifests.bundle_manifest import BundleManifest
+from manifests.test_manifest import TestManifest
+from system.process import Process
+from test_workflow.integ_test.distributions import Distributions
+from test_workflow.test_args import TestArgs
+from test_workflow.test_recorder.test_recorder import TestRecorder
+
+
+class SmokeTestClusterOpenSearch():
+    # dependency_installer: DependencyInstallerOpenSearch
+    repo: GitRepository
+
+    def __init__(
+        self,
+        args: TestArgs,
+        work_dir: str,
+        test_recorder: TestRecorder
+    ) -> None:
+        self.args = args
+        self.work_dir = work_dir
+        self.test_recorder = test_recorder
+        self.process_handler = Process()
+        self.test_manifest = TestManifest.from_path(args.test_manifest_path)
+        self.product = self.test_manifest.name.lower().replace(" ", "-")
+        self.path = args.paths.get(self.product)
+        self.build_manifest = BuildManifest.from_urlpath(os.path.join(self.path, "builds", f"{self.product}", "manifest.yml"))
+        self.bundle_manifest = BundleManifest.from_urlpath(os.path.join(self.path, "dist", f"{self.product}", "manifest.yml"))
+        self.version = self.bundle_manifest.build.version
+        self.platform = self.bundle_manifest.build.platform
+        self.arch = self.bundle_manifest.build.architecture
+        self.dist = self.bundle_manifest.build.distribution
+        self.distribution = Distributions.get_distribution(self.product, self.dist, self.version, work_dir)
+
+    def cluster_version(self) -> str:
+        return self.version
+
+    def download_or_copy_bundle(self, work_dir: str) -> str:
+        extension = "tar.gz" if self.dist == "tar" else self.dist
+        artifact_name = f"{self.product}-{self.version}-{self.platform}-{self.arch}.{extension}"
+        src_path = '/'.join([self.path.rstrip("/"), "dist", f"{self.product}", f"{artifact_name}"]) \
+            if self.path.startswith("https://") else os.path.join(self.path, "dist",
+                                                                  f"{self.product}", f"{artifact_name}")
+        dest_path = os.path.join(work_dir, artifact_name)
+
+        if src_path.startswith("https://"):
+            logging.info(f"Downloading artifacts to {dest_path}")
+            response = requests.get(src_path)
+            with open(dest_path, "wb") as file:
+                file.write(response.content)
+        else:
+            logging.info(f"Trying to copy {src_path} to {dest_path}")
+            # Only copy if it's a file
+            if os.path.isfile(src_path):
+                shutil.copy2(src_path, dest_path)
+                print(f"Copied {src_path} to {dest_path}")
```

Comment:
Replace with logging instead of printing

Reply:
Updated. Thanks.